### PR TITLE
163383153  Implement functionality to report an article

### PR DESCRIPTION
**Pull request by oma0256**
**Description**
If a user finds an article is offensive in one way or another of if it violates the rules of the Authors Haven then they can decide to report this article.

**Type of change**
Feature

**Relevant Stories**
[163383153](https://www.pivotaltracker.com/n/projects/2240000/stories/163383153)

### CODE REVIEW

File: src/containers/Articles/Article/ReportArticleContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        reason: ''
+    };
+
+    onChangeHandler = (e, { value }) => this.setState({ reason: value });
+
+    onSubmitHandler = e => {
+        e.preventDefault();
+        const { 
+            reportArticle, 
+            match: { params: { slug } = {} } = {},
```
Comment (karuhanga) [inline src/containers/Articles/Article/ReportArticleContainer.jsx:18]: @oma0256 this destructuring is hard to follow.

File: src/containers/Articles/Article/ReportArticleContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        } = this.props;
+        const { reason } = this.state;
+        const token = localStorage.getItem('token');
+        const data = { reason }; 
+        reportArticle({ token, data, slug, push });
+    };
+
+    render() {
+        const { reason } = this.state;
+        const { match: { params: { slug } = {} } = {} } = this.props;
```
Comment (karuhanga) [inline src/containers/Articles/Article/ReportArticleContainer.jsx:29]: Here as well.

**Review by karuhanga — COMMENTED** [review]
Very good work on this @oma0256.
Would it be possible to have the flag icon on the same line as the rate component?
Also,

**Comment by oma0256** [comment]
> Very good work on this @oma0256.
> Would it be possible to have the flag icon on the same line as the rate component?
> Also,

That would be fixed immediately @FahdJamy gets merged as he added a column for where I can insert it

File: src/containers/Articles/Article/ReportArticleContainer.jsx (hunk 1 above)
Comment (missvicki) [reply src/containers/Articles/Article/ReportArticleContainer.jsx:18]: @collinewait had advised me to use the same on my previous PR,` const { match: {params: {slug}} = {params: {}} } = this.props;` I think it is okay.

File: src/containers/Articles/Article/ReportArticleContainer.jsx (hunk 2 above)
Comment (missvicki) [reply src/containers/Articles/Article/ReportArticleContainer.jsx:29]: I think it looks okay.

**Review by missvicki — APPROVED** [review]

**Review by codjoero — CHANGES_REQUESTED** [review]
Hey @oma0256 , good work here, just a few things I have noticed.
An unauthenticated user can't view any article, link redirects back to main articles list
`rules` link stays loading heads no where.

**Comment by oma0256** [comment]
> Hey @oma0256 , good work here, just a few things I have noticed.
> An unauthenticated user can't view any article, link redirects back to
> `rules` link stays loading heads no where.

@codjoero thanks for the notice this has been fixed

**Review by codjoero — APPROVED** [review]
Thanks for the changes

**Review by missvicki — APPROVED** [review]

**Review by fahdjamy — APPROVED** [review]
